addons/website/static/src/js/text_processing.js

```diff
@@ -289,9 +290,15 @@ export function drawTextHighlightSVG(textEl, highlightID) {
  * @param {String} highlightID
  */
 export function applyTextHighlight(topTextEl, highlightID) {
+    const endHighlightUpdate = () =>
+        topTextEl.dispatchEvent(new Event("text_highlight_added", { bubbles: true }));
     // Don't reapply the effects to a highlighted text.
-    if (topTextEl.querySelector(".o_text_highlight_item")) {
-        return;
+    // If the target is invisible, we still need to notify the public widget
+    // that a highlight was detected (It's needed anyway, so the public widget
+    // can link the element to its observer, which tracks size changes and
+    // adapts the highlights accordingly.).
+    if (topTextEl.querySelector(".o_text_highlight_item") || !isVisible(topTextEl)) {
+        return endHighlightUpdate();
```

Comment:
Here you're returning `endHighlightUpdate()`

Reply:
> Here you're returning `endHighlightUpdate()`
> But not here... It seems weird, even if it works. :)

We `return` to prevent the rest of the adaptation code from being executed, since we don't want to apply the effect to **invisible or already highlighted content** (but we also need to notify the public widget that a "highlight content" was detected, so it can keep tracking it) :+1: